backend/app/controllers/eventsController/socket/onConnect.js

```diff
@@ -1,35 +1,42 @@
-            console.log("Owner from restaurant " + restaurant.name + " connected!");
-            socket.join(`${socket.handshake.query.restaurantId}:owner`);
-            console.log(`Connected owner to room: "${socket.handshake.query.restaurantId}:owner"`);
+	if(socket.handshake.query.restaurantId){
+		let restaurant;
+		try {
+			restaurant = await Restaurants.findById(socket.handshake.query.restaurantId);
+		} catch (error) {
+			console.log(error);
+		}
+		if(restaurant){
+			//todo: implement proper authentication of owner
+			if(socket.handshake.query.owner === 'true') {
+				console.log('Owner from restaurant ' + restaurant.name + ' connected!');
+				socket.join(`${socket.handshake.query.restaurantId}:owner`);
+				console.log(`Connected owner to room: "${socket.handshake.query.restaurantId}:owner"`);
 
-        } else if (socket.handshake.query.tabId != null){
-            console.log("Client from restaurant " + restaurant?.name + " connected!");
+			} else if (socket.handshake.query.tabId != null){
+				console.log('Client from restaurant ' + restaurant?.name + ' connected!');
             
-            socket.join(`${socket.handshake.query.restaurantId}:globalNotifications`);
-            console.log(`Connected client to room (Global Notifications): "${socket.handshake.query.restaurantId}:globalNotifications"`);
+				socket.join(`${socket.handshake.query.restaurantId}:globalNotifications`);
+				console.log(`Connected client to room (Global Notifications): "${socket.handshake.query.restaurantId}:globalNotifications"`);
 
-            socket.join(`${socket.handshake.query.restaurantId}:${socket.handshake.query.tabId}`);
-            console.log(`Connected client to room (Tab): "${socket.handshake.query.restaurantId}:${socket.handshake.query.tabId}"`);
+				socket.join(`${socket.handshake.query.restaurantId}:${socket.handshake.query.tabId}`);
+				console.log(`Connected client to room (Tab): "${socket.handshake.query.restaurantId}:${socket.handshake.query.tabId}"`);
 
-        } else {
-            try {
-                delete socket.handshake.query.transport;
-                delete socket.handshake.query.EIO;
-                delete socket.handshake.query.t;
-            } catch (e) {
-                console.log("Error deleting query params from socket handshake: " + e);
-            }
-            console.log("Disconnected client that had the following query: " + JSON.stringify(socket.handshake.query));
-            socket.disconnect();
-        }
-    }
+			} else {
+				try {
+					delete socket.handshake.query.transport;
+					delete socket.handshake.query.EIO;
+					delete socket.handshake.query.t;
+				} catch (e) {
+					console.log('Error deleting query params from socket handshake: ' + e);
+				}
+				console.log('Disconnected client that had the following query: ' + JSON.stringify(socket.handshake.query));
+				socket.disconnect();
+			}
+		}
+
+	}
 
 };
\ No newline at end of file
```

Comment:
The code patch seems to be improving the existing code. Here are some observations:

- The indentation style is inconsistent and may confuse developers.
- The try-catch block is added when calling the Restaurants.findById() method, which is a good practice to handle errors that could occur while querying the database.
- A check is added to see if the restaurant exists before proceeding with further operations. This is a good approach for avoiding null or undefined errors.
- The owner/client logs and room joins seem to be implemented correctly, considering the current use case. However, proper authentication for the owner should be implemented in the future.
- In the else section, deleting unnecessary query params from socket handshake is being done, which is okay as it helps avoid leaking sensitive information.
- Overall, there don't seem to be any bug risks at the moment. However, depending on how this code will be used, there are still areas that can be improved.